druid/src/test/java/org/apache/calcite/adapter/druid/DruidQueryFilterTest.java

```diff
@@ -0,0 +1,144 @@
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.math.BigDecimal;
+import java.util.List;
+
+/**
+ * Test class for druid filter generation
+ */
+public class DruidQueryFilterTest {
+
+    private JavaTypeFactory typeFactory;
+    private RexBuilder rexBuilder;
+    private DruidTable druidTable;
+    private RelDataType rowType;
+    private  DruidQuery.Translator translator;
+
+    @Before
+    public void setup() {
+        typeFactory = new JavaTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+        rexBuilder = new RexBuilder(typeFactory);
+        druidTable = new DruidTable(
+                Mockito.mock(DruidSchema.class),
+                "dataSource",
+                null,
+                ImmutableSet.<String>of(),
+                "timestamp",
+                null
+        );
+        final RelDataType varCharType = typeFactory.createSqlType(SqlTypeName.VARCHAR);
+        rowType = typeFactory.builder()
+                .add("dimensionName", varCharType)
+                .build();
+        translator = new DruidQuery.Translator(druidTable, rowType);
+    }
+
+    @Test
+    public void testInFilter()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException,
+            IOException {
+        final List<? extends RexNode> listRexNodes = Lists.newArrayList(
+                rexBuilder.makeInputRef(rowType, 0),
+                rexBuilder.makeExactLiteral(BigDecimal.valueOf(1)),
+                rexBuilder.makeExactLiteral(BigDecimal.valueOf(5)),
+                rexBuilder.makeLiteral("value1")
+        );
+
+        RexNode inRexNode = rexBuilder.makeCall(SqlStdOperatorTable.IN, listRexNodes);
+        RexNode[] rexNodes = new RexNode[1];
+        rexNodes[0] = inRexNode;
+        Method translateFilter = DruidQuery.Translator.class.getDeclaredMethod(
+                "translateFilter",
+                RexNode.class
+        );
+        translateFilter.setAccessible(true);
+        DruidQuery.JsonInFilter returnValue = (DruidQuery.JsonInFilter) translateFilter
+                .invoke(translator, rexNodes);
+        JsonFactory jsonFactory = new JsonFactory();
+        final StringWriter sw = new StringWriter();
+        JsonGenerator jsonGenerator = jsonFactory.createGenerator(sw);
+        returnValue.write(jsonGenerator);
+        jsonGenerator.close();
+        Assert.assertEquals(
+                "{\"type\":\"in\",\"dimension\":\"dimensionName\",\"values\":[\"1\",\"5\",\"value1\"]}",
+                sw.toString()
+        );
+    }
+
+    @Test
+    public void testBetweenFilter()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException,
+            IOException {
+
+        final List<? extends RexNode> listRexNodes = Lists.newArrayList(
+                rexBuilder.makeInputRef(rowType, 0),
```

Comment:
On reflection, the way we should be doing this is translating RexNode to SqlNode. (See SqlImplementor, used by the JDBC adapter and RelToSqlConverter.) SQL is a richer language than RexNode, and close enough to Druid's expression language.

It's too late to do that now, but we need to do much more work in this area, that's the approach I will be advocating.